src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -321,13 +320,35 @@ public Uri ResolveOperationDefinitionUrl(string operationName)
                     throw new OperationNotImplementedException(string.Format(Resources.OperationNotImplemented, operationName));
             }
 
-            string uriString = UrlHelper.RouteUrl(
+            return GetRouteUri(
+                ActionContext.HttpContext,
                 routeName,
-                values: null,
+                null,
                 Request.Scheme,
                 Request.Host.Value);
+        }
 
-            return new Uri(uriString);
+        private Uri GetRouteUri(HttpContext httpContext, string routeName, RouteValueDictionary routeValues, string scheme, string host)
+        {
+            try
+            {
+                var uriString = UrlHelper.RouteUrl(
+                    routeName,
+                    routeValues,
+                    scheme,
+                    host);
+                return new Uri(uriString);
+            }
+            catch
```

Comment:
Is there a specific type of exception we should handle here?

Reply:
Good Point! Yes this is to handle InvalidOperationException. Updated the code.